Network/Units/Client/Client.cpp

```diff
@@ -6,3 +6,138 @@
 */
 
 #include "Client.hpp"
+
+#include <iostream>
+#include <iomanip>
+
+namespace ABINetwork
+{
+
+Client::Client()
+{
+    _roomPassword = "";
+    _loginState = LoginState::NONE;
+    _serverIp = "";
+    _tcpPort = 0;
+    _udpPort = 0;
+    _io_context = std::make_shared<asio::io_context>();
+    if (!_io_context)
+        throw ABIError("Failed to create the client");
+    _tcp_socket = std::make_shared<asio::ip::tcp::socket>(*_io_context);
+    _udp_socket = std::make_shared<asio::ip::udp::socket>(*_io_context);
+    if (!_tcp_socket || !_udp_socket)
+        throw ABIError("Failed to create the client");
+    _server_endpoint = nullptr;
+    _messageId = 0x0000;
+    _token = 0;
+    _io_thread = nullptr;
+}
+
+Client::~Client()
+{
+    _io_context->stop();
+    if (_io_thread->joinable()) {
+        _io_thread->join();
+    }
+}
+
+bool Client::connectToServer(std::string ipServer, int tcp_port)
+{
+    try {
+        _serverIp = ipServer;
+        _tcpPort = tcp_port;
+        asio::ip::tcp::resolver resolver(*_io_context);
+        asio::ip::tcp::resolver::results_type endpoints = resolver.resolve(_serverIp, std::to_string(_tcpPort));
+        asio::connect(*_tcp_socket, endpoints);
+
+        asio::error_code ec;
+        asio::read(*_tcp_socket, asio::buffer(&_udpPort, sizeof(_udpPort)), ec);
+        asio::read(*_tcp_socket, asio::buffer(&_token, sizeof(_token)), ec);
+        // asio::read(*_tcp_socket, asio::buffer(&_idxPlayerComponent, sizeof(_idxPlayerComponent)), ec);
+        std::cout << "Connected to server with token: " << "0x" << std::hex << std::setw(8) << std::setfill('0') << _token << std::dec << std::endl;
```

Comment:
Do we assume that our server displays all its client token when connection ? Asking in a security pov

Reply:
Discussed with @Peralban : NO.

Reply:
yes: NO